src/qibo/backends/numpy.py

```diff
@@ -279,11 +279,11 @@ def apply_channel_density_matrix(self, channel, state, nqubits):
     def _append_zeros(self, state, qubits, results):
         """Helper method for collapse."""
         for q, r in zip(qubits, results):
-            state = self.np.expand_dims(state, axis=q)
+            state = self.np.expand_dims(state, q)
             if r:
```

Comment:
```suggestion
            if len(r) == 0:
```
Is this a check for an empty list? Better to make the check explicit for readability.

Reply:
But then what is `r` and what are we checking there? I would say it's probably checking whether it is `None` at this point, but better to make sure and in case write it explicitely.

Reply:
def _append_zeros(self, state, qubits, results):
        """Helper method for collapse."""
        for q, r in zip(qubits, results):
            state = self.np.expand_dims(state, q)
>           if len(r):
E           TypeError: object of type 'numpy.intc' has no len()

src\qibo\backends\numpy.py:283: TypeError

Reply:
I tested locally. This is checking if `r` is `0` or `1`. For `r=0`, the `if r` returns `False`. For `r=1`, the `if r` returns `True`.